nautobot_ssot_ipfabric/diffsync/adapter_nautobot.py

```diff
@@ -122,7 +122,9 @@ def load_device(self, filtered_devices: List, location):
                 diffsync=self,
                 name=device_record.name,
                 model=str(device_record.device_type),
-                role=str(device_record.device_role) if str(device_record.device_role) else DEFAULT_DEVICE_ROLE,
+                role=str(device_record.device_role.cf["ipfabric_type"])
+                if str(device_record.device_role.cf["ipfabric_type"])
```

Comment:
For previous sync'ed objects:

`if "ipfabric_type" in device_record.device_role.cf and str(device_record.device_role.cf["ipfabric_type"])`

Reply:
This made it run successfully.

Reply:
@alhogan can you take a look at @justinjeffery-ipf updates? Aiming to get a release out on the 26th.

Reply:
@justinjeffery-ipf are you saying the custom field "ipfabric_type" didn't exists for existing objects in Nautobot? Did you run "nautobot-server post_upgrade" after installing the update?

Reply:
Yes I am pretty sure I did:
``` 
 329  gh pr checkout 131
  334  nautobot-server post_upgrade
  335  nano nautobot-plugin-ssot-ipfabric/nautobot_ssot_ipfabric/diffsync/adapter_nautobot.py
```

Reply:
`obj.cf` just doesn't behave as expected. Fields with `null` values are excluded in the dictionary that is returned.

Reply:
`obj.cf` is just [an alias](https://github.com/nautobot/nautobot/blob/develop/nautobot/extras/models/customfields.py#L134-L139) to `obj._custom_field_data` - it doesn't perform any modification of the dictionary that I'm aware of.